lib/capybara/selector/definition/button.rb

```diff
@@ -12,14 +12,17 @@
       locator_matchers = XPath.attr(:id).equals(locator) |
                          XPath.attr(:name).equals(locator) |
                          XPath.attr(:value).is(locator) |
-                         XPath.attr(:title).is(locator)
+                         XPath.attr(:title).is(locator) |
+                         (XPath.attr(:id) == XPath.anywhere(:label)[XPath.string.n.is(locator)].attr(:for))
+
       locator_matchers |= XPath.attr(:'aria-label').is(locator) if enable_aria_label
       locator_matchers |= XPath.attr(test_id) == locator if test_id
 
       input_btn_xpath = input_btn_xpath[locator_matchers]
 
       btn_xpath = btn_xpath[locator_matchers |
                             XPath.string.n.is(locator) |
+                            XPath.descendant(:label)[XPath.string.n.is(locator)] |
```

Comment:
@twalpole I've added this line as part of locator resolution, but the suite is failing.

I have to admit, I struggle with composing XPath declarations, and would really appreciate any guidance you could provide.